### DISCUSSION TITLE
Compiling tutorial program under Ubuntu Linux

### DISCUSSION THREAD

**Discussion by msmathcomp** [BrunoLevy/geogram#56] [Q&A] [answered]
I followed the instructions from the [Wiki](https://github.com/BrunoLevy/geogram/wiki/compiling_Linux) to compile geogram under Linux. It is currently sitting in `/home/martin/geogram/`. It is built under `~/geogram/build/Linux64-gcc-dynamic-Release/` and there, in the `bin` folder, I can execute the programs.

Now, in a different directory, say `/home/martin/test/` I want to compile the [hello_geogram](https://github.com/BrunoLevy/geogram/tree/main/src/tutorial/01_basics/hello_geogram) tutorial code (ultimately, my own little geogram-based program, but the tutorial is fine for now). However, when I call `cmake .` in the test folder, I get the following error:

`CMake Error at /usr/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Geogram (missing: GEOGRAM_LIBRARY GEOGRAM_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  FindGeogram.cmake:61 (find_package_handle_standard_args)
  CMakeLists.txt:13 (find_package)`

Sure, because there are no environment path variables pointing to geogram. Hence, I tried adding `set(GEOGRAM_INSTALL_PREFIX "/home/martin/")` to the CMakeLists to tell it where to look, but with no effect, the error persists. In the CMakeLists, it states that 

`# Instructions to compile this program`
`# are explained in tutorials/README.txt`

However, [that file](https://github.com/BrunoLevy/geogram/blob/main/src/tutorial/README.txt) is simply empty. So I'm stuck here. What is the easiest way to compile my own geogram program after having compiled geogram?

**Comment by sebmestrallet** [comment, marked as answer]
I had to dig into [Graphite](https://github.com/BrunoLevy/GraphiteThree)'s config files, which is based on geogram.

Going back to the first commits of my geogram-based program, I created an empty project, with geogram as submodule, in [LIHPC-Computational-Geometry/geogram-based_program](https://github.com/LIHPC-Computational-Geometry/geogram-based_program)

I hope I will help you!
